File: src/main/csharp/Serving/Serving/BinaryMessageHandler.cs
using System.IO;

namespace Serving
{
    public abstract class BinaryMessageHandler<S> : MessageHandler<S, JavaBinaryReader>
        where S : ConnectionToServerHandler
    {
        protected override JavaBinaryReader Interpret(byte[] bytes)
        {
            var stream = new MemoryStream(bytes);
            return new JavaBinaryReader(stream);
        }
    }
}

File: src/main/java/hoten/serving/message/BinaryMessageHandler.java
package hoten.serving.message;

import hoten.serving.SocketHandler;
import java.io.ByteArrayInputStream;
import java.io.DataInputStream;

public abstract class BinaryMessageHandler<S extends SocketHandler> extends MessageHandler<S, DataInputStream> {

    @Override
    public DataInputStream interpret(byte[] bytes) {
        return new DataInputStream(new ByteArrayInputStream(bytes));
    }
}
